sysutils/apcupsd/src/opnsense/mvc/app/models/OPNsense/Apcupsd/Apcupsd.xml

```diff
@@ -0,0 +1,163 @@
+<model>
+    <mount>//OPNsense/apcupsd</mount>
+    <version>0.0.2</version>
+    <description>APC UPS configuration</description>
+    <items>
+        <!-- Container -->
+        <general>
+            <!-- field -->
+            <Enabled type="BooleanField">
+                <default>0</default>
+                <Required>Y</Required>
+            </Enabled>
+            <UPSName type="TextField">
```

Comment:
UPSName needs a proper validation as it is not quoted in config, a space in the name might lead to an error

Reply:
Ok, i though <xxx></xxx>  was automatically quoting it as " ". Is this not the case ?
//Dan Lundqvist (mrzaz)

Reply:
No, then you'd need to add this to templating:

`UPSNAME "{{ OPNsense.apcupsd.general.UPSName }}"`

Reply:
Nicola [xbb](https://github.com/xbb):
Could you adjust the ...\service\templates\OPNsense\Apcupsd\apcupsd.conf and add double quotes on all text string items like:

`UPSNAME "{{ OPNsense.apcupsd.general.UPSName }}"`
`UPSCABLE "{{ OPNsense.apcupsd.general.UPSCable }}"`
and so on.

Have some issues with my Git right now so can't check-in anything at the moment.
//Dan Lundqvis ([mrzaz](https://github.com/mrzaz))

Reply:
So I just tested this and it seems like the service doesn't handle quoted strings, it just takes whatever is after the key

![image](https://user-images.githubusercontent.com/239074/152440667-71c2ab25-0183-4f95-a521-fd3974e23e44.png)

![image](https://user-images.githubusercontent.com/239074/152440788-716f2a27-f2ef-45f0-851e-3ed1777c2f2b.png)

![image](https://user-images.githubusercontent.com/239074/152440589-3450b60b-ddc0-4ac2-826d-112db81cfcb9.png)

Reply:
UPSNAME is optional, in some case it's also read from the device itself and setting it does nothing. 

Also in the apcupsd configuration sample file the comment says 8 character max but that's not true, it's only in the case you want to program the eeprom. In fact the man page doesn't say about this limit.

So I've checked the apcupsd source, the name length is 99 characters and it's truncated to 8 if you program the UPS.
If a name is not set and the device doesn't report it, the hostname or "default" is used instead.

Anyway I added validation for only alphanumeric, space, dot, dash and underscore max 99 characters